src/arch/xtensa/smp/cpu.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2018, Intel Corporation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the name of the Intel Corporation nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Tomasz Lauda <tomasz.lauda@linux.intel.com>
+ *
+ */
+
+/**
+ * \file arch/xtensa/smp/cpu.c
+ * \brief Xtensa SMP CPU implementation file
+ * \authors Tomasz Lauda <tomasz.lauda@linux.intel.com>
+ */
+
+#include <arch/alloc.h>
+#include <arch/atomic.h>
+#include <arch/cpu.h>
+#include <arch/idc.h>
+#include <platform/platform.h>
+#include <sof/lock.h>
+#include <sof/schedule.h>
+
+static uint32_t active_cores_mask = 0x1;
+static spinlock_t lock = { 0 };
+
+void arch_cpu_enable_core(int id)
+{
+	uint32_t flags;
+
+	spin_lock_irq(&lock, flags);
+
+	if (!(active_cores_mask & (1 << id))) {
+		/* allocate resources for core */
+		alloc_core_context(id);
+
+		/* enable IDC interrupt for the the slave core */
+		idc_write(IPC_IDCCTL, id,
```

Comment:
This is not xtensa generic code but platform specific (the interrupts are probably better enabled by arch_cpu_enable_core() caller).